core/workspace_svg.js

```diff
@@ -597,6 +627,11 @@ Blockly.WorkspaceSvg.prototype.dispose = function() {
     this.zoomControls_ = null;
   }
 
+  if (this.cursor_) {
```

Comment:
Also dispose of the marker if it exists.

And you don't have to set the marker or cursor to null (we're not going to do that for new stuff, and will phase it out in older stuff).

Reply:
Actually--if the cursor exists, call dispose on it in the Workspace dispose function and not here.  And the marker dispose has to be called here.